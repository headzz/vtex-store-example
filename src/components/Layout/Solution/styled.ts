import { colors } from 'configuration'
import styled from 'styled-components'

export const SolutionStyled = styled.section`
  background-color: ${colors.WHITE};
  padding: 5.6rem 0;
  width: 100%;

  @media (min-width: 768px) {
    padding: 5.6rem 20.8rem;
  }
`
