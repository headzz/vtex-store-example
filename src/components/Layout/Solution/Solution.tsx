import { SolutionStyled } from './styled'

/**
 * @component
 * @name Solution
 * @description Defines the container for the layout of Solution.
 * @namespace components components/Layout/Solution
 * @param {element} children Component(s) that may appear inside the layout.
 *
 */

const Solution: React.FC = ({ children }) => {
  return <SolutionStyled>{children}</SolutionStyled>
}

export default Solution
