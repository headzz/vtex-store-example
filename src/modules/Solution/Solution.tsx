import Layout from 'components/Layout'
import Card from './components/Card'
import { CardContainer } from './styled'

import TextArea from './components/TextArea'

/**
 * @component
 * @name Solution
 * @description creates Solution component section to use on screen.
 * @namespace components modules/Solution
 *
 */
const Solution: React.FC = () => (
  <Layout.Solution>
    <TextArea />
    <CardContainer>
      <Card />
    </CardContainer>
  </Layout.Solution>
)

export default Solution
