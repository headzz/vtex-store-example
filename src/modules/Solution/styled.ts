import styled from 'styled-components'

export const CardContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;

  div {
    margin-bottom: 1.6rem;
  }

  @media (min-width: 768px) {
    flex-direction: row;
    justify-content: space-between;
  }
`
