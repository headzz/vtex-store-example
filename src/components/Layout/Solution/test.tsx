import { render } from '@testing-library/react'

import Solution from './Solution'

describe('<Solution />', () => {
  it('should render component that wraps Solution on page', () => {
    const { container } = render(<Solution />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
