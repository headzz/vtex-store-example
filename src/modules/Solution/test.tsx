import { render } from '@testing-library/react'

import Solution from './Solution'

describe('Solution', () => {
  it('match snapshot', () => {
    const { container } = render(<Solution />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
