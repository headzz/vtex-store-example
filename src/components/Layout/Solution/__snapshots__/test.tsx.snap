// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<Solution /> should render component that wraps Solution on page 1`
] = `
.c0 {
  background-color: #ffffff;
  padding: 5.6rem 0;
  width: 100%;
}

@media (min-width:768px) {
  .c0 {
    padding: 5.6rem 20.8rem;
  }
}

<section
  class="c0"
/>
`
